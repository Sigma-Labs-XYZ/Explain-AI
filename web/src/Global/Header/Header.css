@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@200&family=Tilt+Warp&display=swap");
header {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #000;
  height: 70px;
  width: 100%;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  top: 0;
  z-index: 100;
}

header a {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 30px;
}

@media (min-width: 1000.5px) {
  header a {
    margin-left: 15%;
  }
}

@media (max-width: 550.5px) {
  header a {
    height: 70%;
  }
}

header a img {
  display: block;
  overflow: visible;
  height: 60%;
}

header a img:hover {
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  -webkit-transform: scale(1.05);
  transform: scale(1.05);
}

header div {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: 30px;
}

@media (min-width: 1000.5px) {
  header div {
    margin-right: 15%;
  }
}

header div .btn {
  border: none;
  margin-left: 1%;
  width: 75px;
  height: 30px;
  font-weight: bold;
  font-family: "Tilt Warp", cursive;
  font-size: 19px;
  background-color: #323232;
  border: none;
  color: #a5a4a4;
  font-weight: bolder;
  font-family: "Tilt Warp", cursive;
}

header div .btn.selected {
  background-color: #fff;
  color: #000;
}

@media (max-width: 550.5px) {
  header div .btn {
    display: none;
  }
}

header div .likeIm {
  color: #e3e3e3;
  margin-right: 10px;
  font-size: 16px;
  font-family: "Tilt Warp", cursive;
  min-width: 70px;
  text-align: right;
}

@media (max-width: 330px) {
  header div .likeIm {
    display: none;
  }
}

header div .select-pg {
  width: 75px;
  height: 37px;
  border: none;
  text-align: center;
  font-family: "Tilt Warp", cursive;
  font-size: 18px;
  position: relative;
  padding-right: 5px;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 18 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
  background-repeat: no-repeat, repeat;
  background-position: right 1px top 50%, 0 0;
  background-size: 15px auto, 50%;
}

@media (min-width: 550.5px) {
  header div .select-pg {
    display: none;
  }
}
/*# sourceMappingURL=Header.css.map */
