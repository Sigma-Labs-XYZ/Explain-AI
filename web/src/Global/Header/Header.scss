@import "../../Styling/variables.scss";

header {
  @include flex-row;
  background-color: $black;
  height: 70px;
  width: 100%;
  justify-content: space-between;
  position: fixed;
  top: 0;
  z-index: 100;
  a {
    display: flex;
    height: 100%;
    align-items: center;
    margin-left: 30px;
    @media (min-width: $desktop-width) {
      margin-left: 15%;
    }
    @media (max-width: $phone-width) {
      height: 70%;
    }
    img {
      display: block;
      overflow: visible;
      height: 60%;
      &:hover {
        transition: all 200ms ease-in-out;
        transform: scale(1.05);
      }
    }
  }

  div {
    @include flex-row;
    margin-right: 30px;
    @media (min-width: $desktop-width) {
      margin-right: 15%;
    }
    .btn {
      border: none;
      margin-left: 1%;
      width: 75px;
      height: 30px;
      font-weight: bold;
      font-family: $fontStyle;
      font-size: 19px;
      background-color: $bg-button-inactive;
      border: none;
      color: $btn-text-inactive;
      font-weight: bolder;
      font-family: $fontStyle;

      &.selected {
        background-color: $white;
        color: $black;
      }
      @media (max-width: $phone-width) {
        display: none;
      }
    }
    .likeIm {
      color: $offWhite;
      margin-right: 10px;
      font-size: 16px;
      font-family: $fontStyle;
      min-width: 70px;
      text-align: right;
      @media (max-width: $small-phone-width) {
        display: none;
      }
    }
    .select-pg {
      @include select-style;
    }
  }
}
