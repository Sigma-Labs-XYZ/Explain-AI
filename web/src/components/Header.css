header {
  display: flex;
  align-items: center;
  flex-direction: row;
  background-color: #000;
  justify-content: space-between;
}
header img {
  width: 150px;
  margin: 10px;
  margin-left: 30px;
}
@media (min-width: 1000.5px) {
  header img {
    margin-left: 15%;
  }
}
@media (max-width: 400.5px) {
  header img {
    width: 100px;
  }
}
header div {
  display: flex;
  align-items: center;
  flex-direction: row;
  margin-right: 30px;
}
@media (min-width: 1000.5px) {
  header div {
    margin-right: 15%;
  }
}
header div .btn {
  width: 50px;
  height: 25px;
  margin: 1px;
  background-color: #323232;
  border: none;
  color: #6e6e6e;
  font-weight: bolder;
}
header div .btn.selected {
  background-color: #fff;
  color: #000;
}
@media (max-width: 400.5px) {
  header div .btn {
    display: none;
  }
}
header div p {
  color: #fff;
  margin-right: 10px;
  font-size: smaller;
}
header div .select-pg {
  width: 40px;
  height: 15px;
  border: none;
  text-align: center;
  font-weight: bolder;
  position: relative;
  padding-right: 5px;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 18 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
  background-repeat: no-repeat, repeat;
  background-position: right 1px top 50%, 0 0;
  background-size: 15px auto, 50%;
}
@media (min-width: 400.5px) {
  header div .select-pg {
    display: none;
  }
}
header div .select-adult {
  width: 40px;
  height: 15px;
  border: none;
  text-align: center;
  font-weight: bolder;
  position: relative;
  padding-right: 5px;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 18 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
  background-repeat: no-repeat, repeat;
  background-position: right 1px top 50%, 0 0;
  background-size: 15px auto, 50%;
  font-size: 10px;
}
@media (min-width: 400.5px) {
  header div .select-adult {
    display: none;
  }
} /*# sourceMappingURL=Header.css.map */
