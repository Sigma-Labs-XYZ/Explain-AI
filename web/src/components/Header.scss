$white: #fff;
$black: #000;
$bg-button-inactive: #323232;
$btn-text-inactive: #6e6e6e;

@mixin flex-row {
  display: flex;
  align-items: center;
  flex-direction: row;
}

@mixin select-style {
  width: 40px;
  height: 15px;
  border: none;
  text-align: center;
  font-weight: bolder;
  position: relative;
  padding-right: 5px;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 18 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
  background-repeat: no-repeat, repeat;
  background-position: right 1px top 50%, 0 0;
  background-size: 15px auto, 50%;

  @media (min-width: 400px) {
    display: none;
  }
}

header {
  @include flex-row;
  background-color: $black;
  justify-content: space-between;

  img {
    width: 150px;
    margin: 10px;
    margin-left: 30px;
    @media (min-width: 1000px) {
      margin-left: 15%;
    }
    @media (max-width: 400px) {
      width: 100px;
    }
  }
  div {
    @include flex-row;
    margin-right: 30px;

    @media (min-width: 1000px) {
      margin-right: 15%;
    }
    .btn {
      width: 50px;
      height: 25px;
      margin: 1px;
      background-color: $bg-button-inactive;
      border: none;
      color: $btn-text-inactive;
      font-weight: bolder;
      &:active {
        background-color: $white;
        color: $black;
      }
      @media (max-width: 400px) {
        display: none;
      }
    }
    p {
      color: $white;
      margin-right: 10px;
      font-size: smaller;
    }
    .select-pg {
      @include select-style;
    }
    .select-adult {
      @include select-style;
      font-size: 10px;
    }
  }
}
