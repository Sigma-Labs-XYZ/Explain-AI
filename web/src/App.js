import React, { useEffect } from "react"; // eslint-disable-next-line
import ReactGA from "react-ga4";
import AudienceContext from "./components/AudienceContext";
import AppRouter from "./AppRouter/AppRouter";
import Header from "./Global/Header/Header";
import "./App.css";

ReactGA.initialize(process.env.REACT_APP_MEASUREMENT_ID);

function App() {
  useEffect(() => {
    ReactGA.send({
      hitType: "pageview",
      page: window.location.pathname,
    });
  });
  return (
    <AudienceContext>
      <div className="App">
        <Header />
        <AppRouter />
      </div>
    </AudienceContext>
  );
}

export default App;
