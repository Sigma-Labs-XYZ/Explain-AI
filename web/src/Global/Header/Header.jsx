/* eslint-disable react/jsx-no-bind */
import React from "react";
import { Link } from "react-router-dom";
import "./Header.css";
import DropDown from "./DropDown";
import ButtonSelector from "./ButtonSelector";

export default function Header() {
  const labelText = "Like I'm";

  return (
    <div>
      <header>
        <Link data-test-id="router-Link" to="/" title="Link to homepage">
          <img src="./explainai-logo.png" alt="Explain AI" />
        </Link>
        <div>
          <p>{labelText}</p>
          <ButtonSelector />
          <DropDown />
        </div>
      </header>
    </div>
  );
}
