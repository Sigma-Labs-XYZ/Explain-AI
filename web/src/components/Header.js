import explainailogo from "../images/explainai-logo.png";
import "./Header.scss";
import { useState, useEffect } from "react";

export default function Header() {
  const [age, setAge] = useState(localStorage.getItem("age"));

  useEffect(() => {
    localStorage.setItem("age", age);
  }, [age]);

  return (
    <div>
      <header>
        <img src={explainailogo} alt="logo" />
        <div>
          <p>Like I'm</p>
          <button
            onClick={() => setAge("5")}
            className={age === "5" ? "btn selected" : "btn"}
          >
            5
          </button>
          <button
            onClick={() => setAge("10")}
            className={age === "10" ? "btn selected" : "btn"}
          >
            10
          </button>
          <button
            onClick={() => setAge("Adult")}
            className={age === "Adult" ? "btn selected" : "btn"}
          >
            Adult
          </button>
          <select
            onChange={(e) => setAge(e.target.value)}
            className={age === "Adult" ? "select-adult" : "select-pg"}
            value={age}
          >
            <option value="5">5</option>
            <option value="10">10</option>
            <option value="Adult">Adult</option>
          </select>
        </div>
      </header>
    </div>
  );
}
