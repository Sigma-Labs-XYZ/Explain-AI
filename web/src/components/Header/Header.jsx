/* eslint-disable react/jsx-no-bind */
import React, { useState, useEffect } from "react";
import explainailogo from "../../images/explainai-logo.png";
import "./Header.scss";
import DropDown from "./DropDown";
import ButtonSelector from "./ButtonSelector";

export default function Header() {
  const [age, setAge] = useState(
    localStorage.getItem("age") || localStorage.setItem("age", "5") || "5",
  );

  const labelText = "Like I'm";

  useEffect(() => {
    localStorage.setItem("age", age);
  }, [age]);

  function changeAge(newAge) {
    setAge(newAge);
  }
  return (
    <div>
      <header>
        <img src={explainailogo} alt="logo" />
        <div>
          <p>{labelText}</p>
          <ButtonSelector age={age} changeAge={changeAge} />
          <DropDown age={age} changeAge={changeAge} />
        </div>
      </header>
    </div>
  );
}
