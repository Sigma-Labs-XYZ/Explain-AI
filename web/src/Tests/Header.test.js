import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import Header from "../components/Header";
import App from "../App";

//see if header component renders on first load
test("Header component renders correctly", () => {
  const headerTree = renderer.create(<Header />).toJSON();
  expect(headerTree).toMatchSnapshot();
});

//Test if header changes to displaying Toggle or Dropdown depending on the screensize

test("Header displays Age Toggle if screen size above 400px", () => {
  render(<Header />);
  const button = screen.getAllByRole("button");
  window.resizeTo(300, 300);
  expect(button[0]).not.toBeVisible();
});
//test the buttons

//test the dropdown

//testing screen widths

//test localstorage
// const setLocalStorageMock = (key, value) =>{
//     window.localStorage.setItem(key, value)
// }
describe("localStorage tests", () => {
  beforeEach(() => {
    window.localStorage.clear();
  });
  const key = "age";
  test("should save value to localStorage", () => {
    const value = "10";
    window.localStorage.setItem(key, value);
    expect(localStorage.getItem(key)).toEqual(value);
  });
  test("when age button is clicked it should update localStorage with the correct value", () => {
    render(<Header />);
    const value = "Adult";
    const button = screen.getByRole("button", {
      name: value,
    });
    console.log(button);
    fireEvent.click(button);
    expect(localStorage.getItem(key)).toEqual(value);
  });
  test("data in local storage overwritten when age is changed", () => {
    const newValue = "5";
    window.localStorage.setItem(key, newValue);
    expect(localStorage.getItem(key)).toEqual(newValue);
  });
});

//Check if the logo loads correctly

//Check if the 'Like I'm' <p> appears'
