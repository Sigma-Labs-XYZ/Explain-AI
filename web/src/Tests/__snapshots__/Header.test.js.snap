// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <header>
    <img
      alt="logo"
      src="explainai-logo.png"
    />
    <div>
      <p>
        Like I'm
      </p>
      <button
        className="btn"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        10
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Adult
      </button>
      <select
        className="select-pg"
        onChange={[Function]}
        value={null}
      >
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="adult"
        >
          Adult
        </option>
      </select>
    </div>
  </header>
</div>
`;
